src/autopas/containers/verletClusterLists/VerletClusterCells.h

```diff
@@ -0,0 +1,542 @@
+    _lastTraversalSig = traversalInterface->getSignature();
+  }
+
+  /**
+   * @copydoc VerletLists::deleteHaloParticles
+   */
+  void deleteHaloParticles() override {
+    for (size_t i = 0; i < this->_cells.size(); ++i) {
+      for (size_t j = 0; j < _dummyStarts[i];) {
+        if (not this->_cells[i]._particles[j].isOwned()) {
+          // move to dummy particles
+          auto pos = this->_cells[i]._particles[j].getR();
+          pos[0] += _boxMaxWithHalo[2] + 8 * this->getInteractionLength();
+          this->_cells[i]._particles[j].setR(pos);
+          --_dummyStarts[i];
+          std::swap(this->_cells[i]._particles[j], this->_cells[i]._particles[_dummyStarts[i]]);
+        } else {
+          ++j;
+        }
+      }
+    }
+  }
+
+  /**
+   * @copydoc VerletLists::updateContainer()
+   */
+  std::vector<Particle> updateContainer() override {
+    AutoPasLog(debug, "updating container");
+
+    deleteHaloParticles();
+
+    std::vector<Particle> outsideParticles;
+    for (auto iter = begin(autopas::IteratorBehavior::ownedOnly); iter.isValid(); ++iter) {
+      if (utils::notInBox(iter->getR(), this->getBoxMin(), this->getBoxMax())) {
+        outsideParticles.push_back(*iter);
+        iter.deleteCurrentParticle();
+      }
+    }
+
+    return outsideParticles;
+  }
+
+  bool isContainerUpdateNeeded() const override {
+    if (not _isValid) {
+      return true;
+    }
+    for (size_t i = 0; i < this->_cells.size(); ++i) {
+      size_t pid = 0;
+      const size_t end = (_boundingBoxes[i].size() > 0) ? _boundingBoxes[i].size() - 1 : 0;
+
+      for (size_t cid = 0; cid < end; ++cid)
+        for (unsigned int pic = 0; pic < _clusterSize; ++pic) {
+          if (not particleInSkinOfBox(_boundingBoxes[i][cid], this->_cells[i][pid])) {
+            return true;
+          }
+          ++pid;
+        }
+      for (unsigned int pic = 0; pic < _clusterSize && pid < _dummyStarts[i]; ++pic) {
+        if (not particleInSkinOfBox(_boundingBoxes[i][_boundingBoxes[i].size() - 1], this->_cells[i][pid])) {
+          return true;
+        }
+        ++pid;
+      }
+    }
+
+    return false;
+  }
+
+  TraversalSelectorInfo getTraversalSelectorInfo() const override {
+    return TraversalSelectorInfo(_cellsPerDim, this->getInteractionLength(),
+                                 {_gridSideLength, _gridSideLength, this->getBoxMax()[2] - this->getBoxMin()[2]});
+  }
+
+  ParticleIteratorWrapper<Particle, true> begin(IteratorBehavior behavior = IteratorBehavior::haloAndOwned) override {
+    // Special iterator requires valid structure
+    if (not _isValid) {
+      rebuild();
+    }
+    return ParticleIteratorWrapper<Particle, true>(
+        new internal::VerletClusterCellsParticleIterator<Particle, FullParticleCell<Particle>, true>(
+            &this->_cells, &_dummyStarts, _boxMaxWithHalo[0] + 8 * this->getInteractionLength(), behavior));
+  }
+
+  ParticleIteratorWrapper<Particle, false> begin(
```

Comment:
As long as search regions are not further away from the domain than cutoff + skin

Reply:
what happens in that case?

Reply:
If the container is not in a valid state and the const region iterator with a search region outside the domain where the dummy particles are, is called, it will iterate over dummy Particles. 
Do you prefer min/max the search region to sensible values or throw exception?

Reply:
well, it shouldn't throw an exception, but instead, simply return an invalid iterator.

Reply:
Even for cases with some valid particles in the region? Otherwise thats equivalent to restricting the search region.

Reply:
probably restricting the search region should be fine, as long as you simply don't iterate over dummy particles :)

Reply:
Should getNumParticles include Dummy particles?

Reply:
no!